[WIP] docs updates per issue 18282 cloud installs

londoncalling opened the pull request:
fixes #18282

Signed-off-by: Victoria Bialas victoria.bialas@docker.com

Commit:
fixed links in Digital Ocean example

Signed-off-by: Victoria Bialas <victoria.bialas@docker.com>

londoncalling commented:
@moxiegirl please have a look at this first draft. I removed cloud specific installs, added a generic topic that includes info about using Docker Machine. There are some reviewers notes, and I sent an email to you about the "Cloud" menu, I can't figure out how to get rid of it per identifiers in metadata on index files.

thaJeztah commented on docs/installation/cloud.md:
```diff
@@ -0,0 +1,64 @@
+
+You can install Docker on any cloud platform that runs an operating system (OS) that Docker supports. This includes many flavors and versions of Linux, along with Mac and Windows.
+
+__REVIEWERS NOTE:__ The plan is to remove all the detailed info on each cloud provider that's currently under the "cloud install" topic (https://docs.docker.com/machine/drivers/). Are the following very general steps enough to replace that with?
```
Alternatively, you can either put these notes in the description of your pull request or, if extra attention is needed on a specific part of your pull request, add an inline note on GitHub (like this comment I left here). Doing so prevents these notes from accidentally being merged :smile:

moxiegirl replied:
@thaJeztah Reviewers notes are fairly common in the writer world. I don't have a problem with writer's doing them in preference over inline notes.  They are an advantage to put them in text --- rare is the writer that forgets to take them out.   If it becomes a problem, we can Automate the fix.  

Another compromise would be to put them in comment blocks.  That way they are still here but unlikely to go out.  @londoncalling comment blocks don't have the nice bold tho.

thaJeztah replied:
Thanks for explaining, shows I'm not a writer! :smile:

moxiegirl replied:
No, but you are killer technical editor.

thaJeztah commented:
Thanks @londoncalling! Some hints for working on GitHub;

- I changed the title of the PR, and added "WIP" (work in progress). That gives reviewers an indication it's not finished yet and should not be merged in its current state.
- I edited the PR's description and added `fixes #xxxx`. Doing so will automatically close the related issue when the PR is merged.
- It's good practice to describe the changes in the commit message, so that the goal/intent of the PR can be read in the git history (without having to search the linked issue on GitHub)

I'll leave the initial review to @moxiegirl

moxiegirl commented on docs/installation/cloud.md:
```diff
@@ -0,0 +1,64 @@
+You can install Docker on any cloud platform that runs an operating system (OS) that Docker supports. This includes many flavors and versions of Linux, along with Mac and Windows.
+
+__REVIEWERS NOTE:__ The plan is to remove all the detailed info on each cloud provider that's currently under the "cloud install" topic (https://docs.docker.com/machine/drivers/). Are the following very general steps enough to replace that with?
+
```
@londoncalling I'd add a heading here just to make this section easier to find. Also, having two H2s would emphasize the fact a user can choose between two options.

moxiegirl commented on docs/installation/cloud.md:
```diff
@@ -0,0 +1,64 @@
+
+See [Available driver plugins](https://github.com/docker/machine/blob/master/docs/AVAILABLE_DRIVER_PLUGINS.md) in the docker/machine repo on GitHub.
+
+## Using Docker Machine on Mac and Windows cloud hosts
```
I'd remove this for now.  I don't know that there are any cloud providers that server up Mac instances...and Windows VMs would need to install Toolbox so they could run machine and then they are running Vbox on a Vm in the Cloud to run Docker...which seems an edge case.  @nathanleclaire can speak to that --- I might be totally wrong.

londoncalling commented:
ping @moxiegirl this is the revised version of cloud install that we talked about this morning, please take a look

ping @shykes this is where we are per your request on #18282 , please review

londoncalling commented:
@shykes I pinged you too early, wait for my AWS example to be in here :-/

londoncalling commented:
@moxiegirl , @shykes These files address #18282, and should be ready to merge. Documents installing Docker Engine on cloud providers via manual install or Docker Machine, and gives example walkthroughs of both methods (the first using AWS, the second using Digital Ocean). Streamlines the Install > Docker Engine > Cloud menu item list to 3 topics, and minimizes the need to link out to 3rd party sites for instructions about Docker. (Next, somewhat related step is to improve Machine docs, some of this is done, but more to come.) 

Please let me know if I can merge this pull request, or @moxiegirl feel free to merge if it's good to go.

londoncalling commented:
@moxiegirl @shykes , and for a special holiday treat:  You can go to http://45.55.222.72:8000/engine/installation/cloud/ to review these docs updates being served from the Digital Ocean "Droplet" I created for the example on how to use Docker Machine to provision cloud hosts. :boom: :star2: :sparkles:

thaJeztah commented:
@londoncalling I created optimized versions of your images, and blurred your username in one of the EC2 images :smile:

![screen shot 2015-12-23 at 18 20 39](https://cloud.githubusercontent.com/assets/1804568/11982748/470b32fa-a9ae-11e5-9158-607185ec5ea6.png)

Optimized images are below:
[18627-optimized-images.zip](https://github.com/docker/docker/files/71141/18627-optimized-images.zip)

moxiegirl commented on docs/installation/cloud-ex-aws.md:
```diff
@@ -0,0 +1,196 @@
+
+  * Select a key pair to use for this instance. (When you choose to launch, you need to select a key pair to use. Save the `.pem` file to use in the next steps. Save the )
+
+The instance is now up-and-running. The link path to get back to your EC2 instance on AWS is:
```
"link path" maybe menu path?  I'm not sure putting the path in the code font is working well. It looks like something I cut and paste but it isn't.

moxiegirl commented on docs/installation/cloud-ex-aws.md:
```diff
@@ -0,0 +1,196 @@
+
+      ![Choose a general purpose instance type](images/ec2_instance_type.png)
+
+  * Configure the instance.
```
I don't know how you did it but this displays as a numbered step even though they are formatted as bullets.

moxiegirl commented on docs/installation/cloud-ex-aws.md:
```diff
@@ -0,0 +1,196 @@
+
+  * Click **Review and Launch**.
+
+  * Select a key pair to use for this instance. (When you choose to launch, you need to select a key pair to use. Save the `.pem` file to use in the next steps. Save the )
```
Fragment "Save the" and also maybe just put the parenthetical as a follow on sub para.

moxiegirl commented on docs/installation/cloud-ex-aws.md:
```diff
@@ -0,0 +1,196 @@
+    For our example:
+
+        $ cd ~/Desktop/keys/amazon_ec2
+      $ ssh -i "my-key-pair.pem" ubuntu@xx.xxx.xxx.xxx
```
display is off needs another tab

moxiegirl commented on docs/installation/cloud-ex-aws.md:
```diff
@@ -0,0 +1,196 @@
+        gpg: Total number processed: 1
+        gpg:               imported: 1  (RSA: 1)
+
+4. Create a `docker.list` file, and add an entry for our OS, Ubuntu Trusty 14.04 (LTS). (If we were updating an existing file, we'd delete any existing entries.)
```
My preference is use follow on paras rather than parentheticals.  Provides space between what they have to do versus "information about what you are doing.  I'd be interested to learn what you think.

moxiegirl commented on docs/installation/cloud-ex-machine-ocean.md:
```diff
@@ -0,0 +1,239 @@
+
+2. At a command terminal, start Docker Machine and verify that it's running normally.
+
+  * Use `docker-machine ls` to get a list of Docker Machines and their status.
```
Maybe a heading or a step "Try some docker-machine commands" or some such.

moxiegirl commented on docs/installation/cloud-ex-machine-ocean.md:
```diff
@@ -0,0 +1,239 @@
+            NAME      ACTIVE   DRIVER       STATE     URL                         SWARM
+            default   *        virtualbox   Running   tcp:////xxx.xxx.xx.xxx:xxxx  
+
+  * Run a Docker command to make sure that Docker Engine is also up-and-running.
```
Then, I think this is the final step...I'm saying the numbering/bulletting is odd.

moxiegirl commented on docs/installation/cloud-ex-machine-ocean.md:
```diff
@@ -0,0 +1,239 @@
+
+2. Go to the Digital Ocean console to view the new Droplet.
+
+    ![Droplet in Digital Ocean created with Machine](images/ocean_droplet.png)
```
Are there supposed to be two screen caps?

moxiegirl commented on docs/installation/cloud-ex-machine-ocean.md:
```diff
@@ -0,0 +1,239 @@
+
+#### Understand the defaults and options on the create command
+
+For convenience, `docker-machine` will use sensible defaults for choosing settings such as the image that the server is based on, but you override the defaults using the respective flags (e.g. `--digitalocean-image`). This is useful if, for example, you want to create a cloud server with a lot of memory and CPUs (by default `docker-machine` creates a small server). For a full list of the flags/se ...
```
Wrap...
Also "but you can override" ? 
This is > Overrides are maybe?
I might break this chunky para at the sentence For.

moxiegirl commented:
@londoncalling Few line notes but minor issues all around. HUGE improvement and love the tone.  Thank you for this.  With clean up of minor issues LGTM.

londoncalling commented:
@moxiegirl, @thaJeztah, thanks for the great comments, and optimizing the images. 

* my most recent commit addresses your review

* the updated files are served up at: http://45.55.222.72:8000/engine/installation/cloud/.

You might need to refresh your browser to clear the cache.

Let me know if I can merge these changes.

moxiegirl commented:
@thaJeztah If you are cool with the updates and approve the images. Please given an LGTM and merge.

moxiegirl commented:
@thaJeztah I have LGTM on this and @londoncalling has author LGTM. We are merging absent further comments. Please open issues if you note problems after the fact.

thaJeztah commented:
@moxiegirl oh, changes LGTM, I wanted to ask if we need to create redirects from the "old" instructions to the new "cloud" index

moxiegirl commented:
@thaJeztah Great catch. Yes, we do.  @londoncalling I'll create an issue and assign that to you. Its a tweak of the metadata. I'll show you how next week.

londoncalling commented:
Excellent! Thanks!

On Wed, Dec 30, 2015 at 11:33 AM, moxiegirl <notifications@github.com>
wrote:

> @thaJeztah <https://github.com/thaJeztah> Great catch. Yes, we do.
> @londoncalling <https://github.com/londoncalling> I'll create an issue
> and assign that to you. Its a tweak of the metadata. I'll show you how next
> week.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/18627#issuecomment-168060688>.
>
